packages/core/src/external/sftp/surescripts/message.ts

```diff
@@ -0,0 +1,168 @@
+        patient.data
+      );
+      const gender = patient.data.genderAtBirth ?? "U";
+      const genderAtBirth = gender === "O" ? "U" : gender;
+
+      const address = Array.isArray(patient.data.address)
+        ? patient.data.address[0]
+        : patient.data.address;
+      if (!address) return null;
+
+      return toSurescriptsPatientLoadRow(
+        {
+          recordType: "PNM",
+          recordSequenceNumber: index + 1,
+          assigningAuthority: METRIPORT_OID,
+          patientId: patient.id,
+          lastName,
+          firstName,
+          middleName,
+          prefix,
+          suffix,
+          addressLine1: address.addressLine1,
+          addressLine2: address.addressLine2,
+          city: address.city,
+          state: address.state,
+          zip: address.zip,
+          dateOfBirth: patient.data.dob.replace(/-/g, ""), // TODO: check order
+          genderAtBirth,
+          npiNumber: transmission.npiNumber,
+        },
+        patientLoadDetailSchema,
+        patientLoadDetailOrder
+      );
+    })
+    .filter(Boolean) as Buffer[];
+
+  const footer = toSurescriptsPatientLoadRow(
+    {
+      recordType: "TRL",
+      totalRecords: details.length,
+    },
+    patientLoadFooterSchema,
+    patientLoadFooterOrder
+  );
+
+  return Buffer.concat([header, ...details, footer]);
+}
+
+export function toSurescriptsPatientLoadRow<T extends object>(
+  data: T,
+  schema: z.ZodObject<z.ZodRawShape>,
+  order: OutgoingFileRowSchema<T>
+): Buffer {
+  if (!schema.safeParse(data).success) {
+    console.log("Invalid data", data);
+  }
+  return Buffer.from(order.map(field => field.toSurescripts(data)).join("|"), "ascii");
+}
+
+export function fromSurescriptsVerificationFile(message: Buffer) {
+  const rows = message
+    .toString("ascii")
+    .split("\n")
+    .map(row => row.split("|").map(cell => cell.trim().replace(/\\F\\/g, "|")));
+  const header = rows.shift();
+  const details = rows.slice(0, -1);
+  const footer = rows.pop();
+
+  if (!header) throw new Error("Header is required");
+  if (!details || details.length === 0) throw new Error("Details are required");
+  if (!footer) throw new Error("Footer is required");
+
+  const headerData = fromSurescriptsRow(
+    header,
+    patientVerificationHeaderOrder,
+    isPatientVerificationHeader
+  );
+  const detailsData = details.map(detail =>
+    fromSurescriptsRow(detail, patientVerificationDetailOrder, isPatientVerificationDetail)
+  );
+  const footerData = fromSurescriptsRow(
+    footer,
+    patientVerificationFooterOrder,
+    isPatientVerificationFooter
+  );
+
+  return { header: headerData, details: detailsData, footer: footerData };
+}
+
+function fromSurescriptsRow<T extends object>(
+  row: string[],
+  rowOrder: IncomingFileRowSchema<T>,
+  rowValidator: (data: object) => data is T
+): T {
+  const data: Partial<T> = {};
+  for (const field of rowOrder) {
```

Comment:
nit: row "order" and "field" don't seem to play along... the "order" in the `patientVerificationHeaderOrder`, `patientVerificationDetailOrder` and the like doesn't seem to indicate what's really there. Maybe something like "patientHeaderConversionSettings" or something like that?

Reply:
Nice that makes sense! Will change the names here to be more meaningful.